x-pack/plugins/ml/public/application/notifications/components/ml_entity_selector.tsx

```diff
@@ -0,0 +1,173 @@
+  /**
+   * In case the there are duplicated IDs across different ML entity types,
+   * they should be de-selected simultaneously if this setting is enabled.
+   */
+  handleDuplicates?: boolean;
+}
+
+export const MlEntitySelector: FC<MlEntitySelectorProps> = ({
+  entityTypes = { anomaly_detector: true, data_frame_analytics: true, trained_models: true },
+  multiSelect = true,
+  selectedOptions,
+  onSelectionChange,
+  handleDuplicates = false,
+}) => {
+  const { jobs: jobsApi, trainedModels, dataFrameAnalytics } = useMlApiContext();
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [options, setOptions] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
+
+  const visColorsBehindText = euiPaletteColorBlindBehindText();
+
+  const fetchOptions = useCallback(async () => {
+    try {
+      const newOptions: Array<EuiComboBoxOptionOption<string>> = [];
+      if (entityTypes?.anomaly_detector) {
+        const { jobIds: jobIdOptions } = await jobsApi.getAllJobAndGroupIds();
+
+        newOptions.push({
+          label: i18n.translate('xpack.ml.mlEntitySelector.jobOptionsLabel', {
+            defaultMessage: 'Anomaly detection jobs',
+          }),
+          isGroupLabelOption: true,
+          color: visColorsBehindText[0],
+          options: jobIdOptions.map((v) => ({
+            label: v,
+            value: v,
+            key: `anomaly_detector:${v}`,
+            color: visColorsBehindText[0],
+          })),
+        });
+      }
+
+      if (entityTypes?.data_frame_analytics) {
+        const dfa = await dataFrameAnalytics.getDataFrameAnalytics();
+        if (dfa.count > 0) {
+          newOptions.push({
+            label: i18n.translate('xpack.ml.mlEntitySelector.dfaOptionsLabel', {
+              defaultMessage: 'Data frame analytics',
+            }),
+            isGroupLabelOption: true,
+            options: dfa.data_frame_analytics.map((v) => ({
+              label: v.id,
+              value: v.id,
+              key: `data_frame_analytics:${v.id}`,
+              color: visColorsBehindText[2],
+            })),
+          });
+        }
+      }
+
+      if (entityTypes?.trained_models) {
+        const models = await trainedModels.getTrainedModels();
+        if (models.length > 0) {
+          newOptions.push({
+            label: i18n.translate('xpack.ml.mlEntitySelector.trainedModelsLabel', {
+              defaultMessage: 'Trained models',
+            }),
+            isGroupLabelOption: true,
+            options: models.map((v) => ({
+              label: v.model_id,
+              value: v.model_id,
+              key: `trained_models:${v.model_id}`,
+              color: visColorsBehindText[3],
+            })),
+          });
+        }
+      }
+
+      setOptions(newOptions);
+    } catch (e) {
+      // TODO add error handling
+    }
+    setIsLoading(false);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is there a reason why the full list of dependencies is not being supplied?

Reply:
Fixed in https://github.com/elastic/kibana/pull/142778/commits/1daee3057069f5eaf918f366d935c144679e4778